WalletWasabi.Gui/TransactionWatcher.cs

```diff
@@ -0,0 +1,115 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.Gui
+{
+	class CoinEventArgs : EventArgs
+	{
+		public SmartCoin Coin { get; }
+
+		public CoinEventArgs(SmartCoin coin)
+		{
+			Coin = coin;
+		}
+	}
+
+	class CoinsWatcher
+	{
+		private List<SmartCoin> _trackedCoins;
+		private bool _isSynchronized = false;
+		public EventHandler<CoinEventArgs> CoinReceived;
+
+		public CoinsWatcher()
+		{
+			_trackedCoins = new List<SmartCoin>();
+			OnBestHeightChanged(this, Height.Unknown);
+		}
+
+		public void Start()
+		{
+			Global.IndexDownloader.BestHeightChanged += OnBestHeightChanged;
+			Global.WalletService.Coins.HashSetChanged += OnCoinsArrives;
```

Comment:
I don't like this. This `HashSetChanged` was @danwalmsley's hack when we were in that great hurry in Lisbon.  
Also maybe we don't want to mess with confirmed coins, rather we could only watch the mempool? Feels less intrusive for me. Anyway, these are just ideas. When this PR is not WIP anymore I'll look into this.